### PR TITLE
Add support for custom language provided by third-party

### PR DESCRIPTION
**Pull request by itscaro**
## Description

Add template for PHP in order to support built-in build for PHP

## Motivation and Context
- [x] I have raised an issue to propose this change ( #55 )

## How Has This Been Tested?
$ ./build_samples.sh && ./deploy_samples.sh

## Types of changes
- [x] New feature (non-breaking change which adds functionality)

## Checklist:
- [ ] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I've read the [CONTRIBUTION](https://github.com/alexellis/faas/blob/master/CONTRIBUTING.md) guide
- [x] I have signed-off my commits with `git commit -s`
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by alexellis** [comment]
Ref proposal & additional questions on #55

File: proxy/deploy.go [hunk 1]
```diff
@@ -25,7 +25,9 @@ func DeployFunction(fprocess string, gateway string, functionName string, image
 		fprocessTemplate = "ruby index.rb"
 	} else if language == "csharp" {
 		fprocessTemplate = "dotnet ./bin/Debug/netcoreapp2.0/root.dll"
-	}
+	} else if language == "php" {
+    fprocessTemplate = "php index.php"
```
Comment (kylestev) [inline proxy/deploy.go]: Make sure to run `go fmt`

Comment (alexellis) [reply proxy/deploy.go]: @kylestev do you want to add `gofmt` / a liniting check to the CI script?

**Comment by alexellis** [comment]
Hi @itscaro - we will move forward with this shortly. We just need to test and merge this PR first - #85 

If you'd like to help test #85 then we could get your PR merged faster. Thanks again for being part of the project 👍

**Comment by itscaro** [comment]
@alexellis Just read your message, I will look into it.

**Comment by alexellis** [comment]
When we merge #87 then we'll need a new repo creating for PHP to support that. You could start with it now @itscaro ?

**Comment by itscaro** [comment]
OK I'll do this week

**Comment by itscaro** [comment]
@alexellis I pushed PHP templates to this repo: https://github.com/itscaro/openfaas-template-php

**Comment by ericstoekl** [comment]
Derek add label: revisit

File: builder/build.go [hunk 2]
```diff
@@ -13,41 +13,46 @@ import (
 
 // BuildImage construct Docker image from function parameters
 func BuildImage(image string, handler string, functionName string, language string, nocache bool, squash bool) {
-
 	switch language {
```
Comment (itscaro) [inline builder/build.go]: @alexellis is it possible to do like this?

**Comment by rawkode** [comment]
If I can help with this, please let me know, @itscaro

**Comment by alexellis** [comment]
@itscaro I'm going to close this PR because I think we have already done the work in other PRs. Please feel free to re-submit a new PR with anything that is difference such as unit tests.

**Pull request closed without merging**